src/debug/configurationProvider.ts

```diff
@@ -5,62 +5,66 @@ import * as path from "path";
+
+    constructor() { }
+
+    public provideDebugConfigurations(folder: vscode.WorkspaceFolder | undefined, token?: vscode.CancellationToken):
+        vscode.ProviderResult<vscode.DebugConfiguration[]> {
+        return [{
+            name: "Arduino",
+            type: "arduino",
+            request: "launch",
+            program: "${file}",
+            cwd: "${workspaceRoot}",
+            MIMode: "gdb",
+            targetArchitecture: "arm",
+            miDebuggerPath: "",
+            debugServerPath: "",
+            debugServerArgs: "",
+            customLaunchSetupCommands: [
+                {
+                    text: "target remote localhost:3333",
+                },
+                {
+                    text: "file ${file}",
+                },
+                {
+                    text: "load",
+                },
+                {
+                    text: "monitor reset halt",
+                },
+                {
+                    text: "monitor reset init",
+                },
+            ],
+            stopAtEntry: true,
+            serverStarted: "Info\\ :\\ [\\w\\d\\.]*:\\ hardware",
+            launchCompleteCommand: "exec-continue",
+            filterStderr: true,
+            args: [],
+        }];
     }
 
-    public async run(config) {
-        // Default settings:
-        if (!config.request) {
-            config = {
-                name: "Arduino",
-                type: "arduino",
-                request: "launch",
-                program: "${file}",
-                cwd: "${workspaceRoot}",
-                MIMode: "gdb",
-
-                targetArchitecture: "arm",
-                customLaunchSetupCommands: [
-                    {
-                        text: "target remote localhost:3333",
-                    },
-                    {
-                        text: "file ${file}",
-                    },
-                    {
-                        text: "load",
-                    },
-                    {
-                        text: "monitor reset halt",
-                    },
-                    {
-                        text: "monitor reset init",
-                    },
-                ],
-                stopAtEntry: true,
-                serverStarted: "Info\\ :\\ [\\w\\d\\.]*:\\ hardware",
-                launchCompleteCommand: "exec-continue",
-                filterStderr: true,
-                args: [],
-            };
+    // Try to add all missing attributes to the debug configuration being launched.
+    public resolveDebugConfiguration(folder: vscode.WorkspaceFolder | undefined, config: vscode.DebugConfiguration, token?: vscode.CancellationToken):
+        vscode.ProviderResult<vscode.DebugConfiguration> {
+        return this.resolveDebugConfigurationAsync(config);
+    }
+
+    private async resolveDebugConfigurationAsync(config: vscode.DebugConfiguration) {
+        if (!ArduinoContext.initialized) {

```

Comment:
in resolve method, why not generate the default settings for `if (!config.request)`?